arangod/RocksDBEngine/RocksDBEdgeIndex.h

```diff
@@ -57,10 +57,16 @@ class RocksDBEdgeIndexIterator final : public IndexIterator {
   char const* typeName() const override { return "edge-index-iterator"; }
   bool hasExtra() const override { return true; }
   bool next(LocalDocumentIdCallback const& cb, size_t limit) override;
+  bool nextCovering(DocumentCallback const& cb, size_t limit) override;
   bool nextExtra(ExtraCallback const& cb, size_t limit) override;
   void reset() override;
+  
+  /// @brief we provide a method to provide the index attribute values
+  /// while scanning the index
+  bool hasCovering() const override { return true; }
 
  private:
+  bool initKey(arangodb::velocypack::Slice& key);
```

Comment:
Please add a comment that caller needs to make sure that the Slice data is not deleted as long as this Iterator is ongoing.
The Iterator will only retain the Slice and will not copy out the data.
If this is forgotten we end up in Undefined behaviour.